Include more resource types in the tsh guide

ptgott opened the pull request:
The user guide to tsh currently assumes that the user wants to access a Teleport-protected server. This change edits the page to be a more general-purpose getting started guide for connecting to resources with tsh.

- **Edit the Introduction section.** Name it "Basic usage" and edit it to account for all supported resource types. Show the main flow of using tsh to list and connect to resources.
- **Reduce the number of H2s related to SSH servers.** These sections drown out non-SSH related content. Include them in two H2s: one for connecting to SSH servers and one for interacting with SSH servers. In some modified sections, remove tabs in which the tab items only differ in their Proxy Service addresses.
- **Remove a duplicate section on aliases.** There are two sections on the page with substantially similar content.
- **Tighten up the intro paragraphs** and make them less SSH specific. Also add a link to the Access Request user guide.
- **Modify the node-logins.mdx partial.** Make it about all Teleport resources, since it is not specific to servers.
- **Put the installation section first** since a user will want to follow this before "Basic usage".

ptgott committed:
Include more resource types in the tsh guide

The user guide to tsh currently assumes that the user wants to access a
Teleport-protected server. This change edits the page to be a more
general-purpose getting started guide for connecting to resources with
tsh.

- **Edit the Introduction section.** Name it "Basic usage" and edit it
  to account for all supported resource types. Show the main flow of
  using tsh to list and connect to resources.
- **Reduce the number of H2s related to SSH servers.** These sections
  drown out non-SSH related content. Include them in two H2s: one for
  connecting to SSH servers and one for interacting with SSH servers. In
  some modified sections, remove tabs in which the tab items only differ
  in their Proxy Service addresses.
- **Remove a duplicate section on aliases.** There are two sections on
  the page with substantially similar content.
- **Tighten up the intro paragraphs** and make them less SSH specific.
  Also add a link to the Access Request user guide.
- **Modify the node-logins.mdx partial.** Make it about all Teleport
  resources, since it is not specific to servers.
- **Put the installation section first** since a user will want to
  follow this before "Basic usage".

ravicious commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -6,142 +6,244 @@ labels:
+Windows desktops are available to list in Teleport Connect and the Teleport Web
+UI.
+
+`tsh <resource> ls` commands can apply a filter based on the resource's
+labels.  For example, to only show servers with the `os` label set to `osx`,
+you can run the following command:
```
TIL, I didn't know that it's a widely available syntax for this. ;)

ravicious commented on docs/pages/connect-your-client/tsh.mdx:
```diff
@@ -6,142 +6,244 @@ labels:
 
-# ... and now your 'ssh' command is calling Teleport's `tsh ssh`
-$ ssh user@host
+You can only connect to a Windows desktop using the Teleport Web UI or Teleport
+Connect. To do so, use the left sidebar to navigate to **Resources > Desktops**
+and select a desktop to connect to.
```
Connect doesn't have a sidebar, I guess we could either reword this to say "To do so in the Web UI" or remove the second sentence completely?

ravicious commented on docs/pages/connect-your-client/tsh.mdx:
````diff
@@ -6,142 +6,244 @@ labels:
+ ```code
+ $ tsh apps login grafana
+ ```
 
-Follow the instructions below to install the `tsh` binary.
+You can then access the application with a command like the following:
 
-We recommend installing `tsh` of the same major version as the version used in
-your Teleport cluster.
+```code
+$ curl \
+  --cert /Users/alice/.tsh/keys/teleport.example.com/alice-app/cluster-name/grafana-x509.pem \
+  --key /Users/alice/.tsh/keys/teleport.example.com/alice \
+  https://grafana.teleport.example.com:3080
+```
 
-To find the version number, either:
+For TCP applications, use `tsh` to start a local proxy server and connect to
+your application with your client of choice. This example connects to the app
+`tcp-app`:
 
-- In the Web UI, select your username in the upper right, then click
-  **Help & Support**. You will see the version of your Teleport
-  cluster under **CLUSTER INFORMATION**.
+```code
+$ tsh proxy app tcp-app
+```
````
Both commands work for both types of apps. The only real difference is that the proxy automatically manages the certs and enriches with them connections made through the proxy whereas `tctl apps login` just fetches the cert that you have to them manually use with whichever client that you use.

I'd say in the vast majority of cases, a proxy is preferred, so I think we could first start with it and only later mention `tsh apps login`.

It'd be a good place to link to the VNet docs too since VNet makes this even easier than `tsh proxy app`.

ravicious reviewed (approved):
I found one minor issue with the commands for apps, the rest looks fine to me.

ptgott committed:
Respond to ravicious feedback on the tsh guide

- Remove incorrect sentences indicating that you can navigate using the
  sidebar in Teleport Connect.
- In the section re: connecting to an appliction, move `tsh proxy app`
  before `tsh apps login`. Recommend using `tsh proxy app` and clarify
  the difference between the two approaches.
